File: hw/vaca/u3/MazeGeneratorJava/src/main/java/ec/edu/espe/mazegenerator/model/MazeRoom.java
package ec.edu.espe.mazegenerator.model;
import java.util.EnumSet;

/**
 *
 * @author Kevin Vaca Edison's OOP ESPE
 */
public class MazeRoom {
    public enum Direction { NORTH, SOUTH, EAST, WEST }

    private EnumSet<Direction> openWalls;
    
    public MazeRoom() {
        openWalls = EnumSet.noneOf(Direction.class);
    }

    public void openWall(Direction dir) {
        openWalls.add(dir);
    }

    public boolean isOpen(Direction dir) {
        return openWalls.contains(dir);
    }

}

File: hw/vaca/u3/MazeGeneratorJs/MazeRoom.js
export class MazeRoom {
  static Direction = { NORTH: "NORTH", SOUTH: "SOUTH", EAST: "EAST", WEST: "WEST" };

  constructor() {
    this.openWalls = new Set();
  }

  openWall(dir) {
    this.openWalls.add(dir);
  }

  isOpen(dir) {
    return this.openWalls.has(dir);
  }
}
